code/controllers/subsystem/SSticker.dm

```diff
@@ -607,6 +607,30 @@ SUBSYSTEM_DEF(ticker)
 			var/mob/M = m
 			H.add_hud_to(M)
 
+	var/static/list/base_encouragement_messages = list(
+		"Keep on keeping on!",
+		"Great job!",
+		"Keep up the good work!",
+		"Nice going!",
+	)
+
+	var/static/list/special_encouragement_messages = list(
+		"Outstanding!",
+		"This is going on the fridge!",
+		"Looks like you're popular!",
+		"That's what we like to see!",
+		"Hell yeah, brother!"
+	)
+
+	// Tell people how many kudos they got this round
+	// Besides, what's another loop over the /entire player list/
+	var/kudos_message
+	for(var/mob/M in GLOB.player_list)
+		var/kudos = M.mind?.kudos_received_from
+		if(length(kudos))
+			kudos_message = pick(length(kudos) > 10 ? special_encouragement_messages : base_encouragement_messages)
+			to_chat(M, "<span class='green big'>You received <b>[length(M.mind?.kudos_received_from)]</b> kudos from other players this round! [kudos_message]</span>")
```

Comment:
You check if `mind` exists in line 629 and 630, so I don't think that the `?.` operator in line 632 is necessary
Also, maybe change the amount of kudos to get a special message to 5? I don't think people would be getting 10 kudos often if ever

Reply:
Good points, addressed